pkg/ccl/backupccl/restore_span_covering_test.go

```diff
@@ -1009,13 +1042,81 @@ func TestRestoreEntryCover(t *testing.T) {
 									backups[numBackups-1].Spans,
 									backups,
 									layerToIterFactory,
+									nil,
 									target<<20,
 									introducedSpanFrontier,
 									[]jobspb.RestoreProgress_FrontierEntry{},
 									simpleImportSpans)
 								require.NoError(t, err)
 								require.NoError(t, checkRestoreCovering(ctx, backups, backups[numBackups-1].Spans,
 									cover, target != noSpanTargetSize, execCfg.DistSQLSrv.ExternalStorage))
+
+								// Check that the correct import spans are created if the job is
+								// resumed after the completion of some random entries in the cover.
+								if len(cover) > 0 {
+									for n := 1; n <= 5; n++ {
+										var completedSpans []roachpb.Span
+										var highWater []byte
+										var frontierEntries []jobspb.RestoreProgress_FrontierEntry
+
+										// Randomly choose to use frontier checkpointing instead of
+										// explicitly testing both forms to avoid creating an exponential
+										// number of tests.
+										useFrontierCheckpointing := rand.Intn(2) == 0
+										if useFrontierCheckpointing {
+											completedSpans = getRandomCompletedSpans(cover, n)
+											for _, sp := range completedSpans {
+												frontierEntries = append(frontierEntries, jobspb.RestoreProgress_FrontierEntry{
+													Span:      sp,
+													Timestamp: completedSpanTime,
+												})
+											}
+										} else {
+											idx := r.Intn(len(cover))
+											completedSpans = append(completedSpans, roachpb.Span{
```

Comment:
nit: we shouldn't need to create a list of completed spans for this to work. If restore is still using a hwm, the complete spans list will always be empty.